### ISSUE TITLE
Dynamicly rendered template does not update its helpers when used with a notfound route

### ISSUE THREAD

**Issue by serkandurusoy** [kadirahq/flow-router#243]
I've just posted an issue over at `zimme:active-route` https://github.com/zimme/meteor-active-route/issues/39 but I also *suspect* that this may be a flow-router issue. If it is not, please ignore this. But if it is possible that flow-router is the culprit, here's a rundown:

I separated my main layout into sections, one of which is a navigation template (`sidenav`) and oddly, when I enter a non-existing (using a broken link on the menu, not directly from the address bar) route, which shares the same template, active route helpers on that template don't get updated with the latest route information.

Here's the route definition:

````javascript
FlowRouter.notFound = {
  action: function() {
    BlazeLayout.render('mainLayout', { main: "notFound", navbar: "topNav", sidenav: "sidenav" })
  }
};

FlowRouter.route('/', {
  name: 'home',
  action: function(params) {
    BlazeLayout.render('mainLayout', { main: "home", navbar: "topNav", sidenav: "sidenav" });
  }
});

FlowRouter.route('/contact', {
  name: 'contact',
  action: function(params) {
    BlazeLayout.render('mainLayout', { main: "contact", navbar: "topNav", sidenav: "sidenav" });
  }
});
````

**Comment by arunoda** [comment]
Good question. In this case, when you hit a notFound route, if your is looking for the route via our APIs like `.getParam()` They won't get notified. That's by design.

I need to check whether, we are invalidating `watchPathChange()` even in the notfound route. May be we need to add a special flag on the route context to mention, this is a not found route.

I think this is a feature we should implement.

**Comment by serkandurusoy** [comment]
I think this may also be related to https://github.com/kadirahq/flow-router/issues/188 such that, notFound should be handled just like any other route.

I don't quite understand why this is a *special* route. In fact, this is currently like a catch all route which lacks properties that other routes have.

And since it cannot provide necessary 404 http codes (no server side), then it feels like it should be implemented just like

````javascript
FlowRouter.route('/:notFound*', {
  name: 'notFound',
  action: FlowRouter.notFound.action()
});
````

**Comment by arunoda** [comment]
That's right. But we need to add it after all the routes has been defined. Since we can define routes dynamically, we need to do some special care about that.

I think we'll have a proper fix for this with triggers soon.

**Comment by serkandurusoy** [comment]
Great! Thanks.
